### PR TITLE
Bug fix; custom image storage read

### PR DESCRIPTION
**Pull request by milismsft**
Fix unexpected exception thrown while attempting to retrieve the list of custom images from a specified storage account.

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy [hunk 1]
```diff
@@ -39,43 +39,47 @@ class AzureStorageClient {
...
-          if (blobDirectoryList.size() > 0) {
-            def dir = blobDirectoryList.remove(0)
-            def blob = container.getDirectoryReference(dir)
-
-            while (blobDirectoryList.size() > 0) {
-              dir = blobDirectoryList.remove(0)
-              blob = blob.getDirectoryReference(dir)
-            }
+    imageStorageList?.each {AzureCustomImageStorage storage ->
+      if (storage && storage.scs && storage.blobDir && storage.osType) {
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy]: Ok as-is, but may read better as: `if (storage?.scs && storage?.blobDir && storage?.osType)`.
Or maybe not.
I can live with it either way.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy [hunk 2]
```diff
@@ -39,43 +39,47 @@ class AzureStorageClient {
...
+        try {
+          ArrayList<String> blobDirectoryList = new ArrayList<String>()
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy]: `ArrayList<String> blobDirectoryList = []`

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy [hunk 3]
```diff
@@ -39,43 +39,47 @@ class AzureStorageClient {
...
+          CloudStorageAccount storageAccount = CloudStorageAccount.parse(storage.scs)
+
+          // retrieve the blob client.
+          CloudBlobClient blobClient = storageAccount.createCloudBlobClient()
+          String dirDelimiter = blobClient.getDirectoryDelimiter()
+          blobDirectoryList.addAll(storage.blobDir.split(dirDelimiter))
+          def container = blobClient.getContainerReference(blobDirectoryList.remove(0))
+
+          if (container) {
+            if (blobDirectoryList.size() > 0) {
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy]: `if (blobDirectoryList.size()) {`

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy [hunk 4]
```diff
@@ -39,43 +39,47 @@ class AzureStorageClient {
...
+              def dir = blobDirectoryList.remove(0)
+              def blob = container.getDirectoryReference(dir)
+
+              while (blobDirectoryList.size() > 0) {
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureStorageClient.groovy]: Same comment here.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/security/AzureNamedAccountCredentials.groovy [hunk 5]
```diff
@@ -94,7 +122,7 @@ public class AzureNamedAccountCredentials implements AccountCredentials<AzureCre
 
     public AzureRegion(String name) {
       if (name == null) {
-        throw new NullPointerException("name")
+        throw new IllegalArgumentException("name")
```
Comment (duftler) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/security/AzureNamedAccountCredentials.groovy]: s/`"name"/"name must be specified."

**Comment by duftler** [comment]
Looks like you responded to the feedback. GitHub doesn't send a notification unless you add a comment to the PR thread.
Needs a rebase.
LGTM.

**Comment by milismsft** [comment]
@duftler thanks for looking at this; rebase is done.

**Comment by duftler** [comment]
Looks like quick-draw @lwander beat us to the commit button. Will need another rebase.

**Comment by milismsft** [comment]
rebased one more time ;-)

**Pull request merged**